File: eco-core/core-plugin/src/main/java/com/willfp/ecomc/LevelPlaceholder.java
package com.willfp.ecomc;

import com.willfp.eco.core.integrations.placeholder.PlaceholderManager;
import com.willfp.eco.core.placeholder.PlayerPlaceholder;
import com.willfp.eco.util.StringUtils;
import org.bukkit.potion.PotionEffectType;

public class LevelPlaceholder {
    public static void register() {
        PlaceholderManager.registerPlaceholder(new PlayerPlaceholder(
                EcoMCPlugin.getInstance(),
                "level",
                player -> {
                    int level;
                    try {
                        level = Integer.parseInt(PlaceholderManager.translatePlaceholders("%battlepass_tier%", player));
                    } catch (NumberFormatException e) {
                        return "§80✯";
                    }
                    String color;

                    if (level == 1) {
                        color = "§8";
                    } else if (level < 10) {
                        color = "§7";
                    } else if (level < 20) {
                        color = "§f";
                    } else if (level < 30) {
                        color = "§6";
                    } else if (level < 40) {
                        color = "§3";
                    } else if (level < 50) {
                        color = "§c";
                    } else if (level < 60) {
                        color = "§e";
                    } else if (level < 70) {
                        color = "§9";
                    } else if (level < 80) {
                        color = "§b";
                    } else if (level < 90) {
                        color = "§a";
                    } else if (level < 100) {
                        color = "§d";
                    } else {
                        color = "§a§l";
                    }

                    return color + level + "✯";
                }
        ));

        PlaceholderManager.registerPlaceholder(new PlayerPlaceholder(
                EcoMCPlugin.getInstance(),
                "color",
                player -> {
                    return PlaceholderManager.translatePlaceholders("%luckperms_prefix%", player).substring(0, 2);
                }
        ));

        PlaceholderManager.registerPlaceholder(new PlayerPlaceholder(
                EcoMCPlugin.getInstance(),
                "rank",
                player -> {
                    String prefix = PlaceholderManager.translatePlaceholders("%luckperms_prefix%", player);
                    return prefix.replace(" ", "")
                            .replace("|", "");
                }
        ));

        PlaceholderManager.registerPlaceholder(new PlayerPlaceholder(
                EcoMCPlugin.getInstance(),
                "tag",
                player -> {
                    String tag = PlaceholderManager.translatePlaceholders("%deluxetags_tag%", player);
                    if (tag == null || tag.isEmpty() || tag.isBlank()) {
                        return "";
                    } else {
                        return " " + tag;
                    }
                }
        ));

        PlaceholderManager.registerPlaceholder(new PlayerPlaceholder(
                EcoMCPlugin.getInstance(),
                "heart",
                player -> {
                    if (player.hasPotionEffect(PotionEffectType.ABSORPTION)) {
                        return StringUtils.format("&6❤");
                    } else {
                        return StringUtils.format("&c❤");
                    }
                }
        ));
    }
}

File: eco-core/core-plugin/src/main/kotlin/com/willfp/ecomc/LevelPlaceholder.kt
package com.willfp.ecomc

import com.willfp.eco.core.EcoPlugin
import com.willfp.eco.core.integrations.placeholder.PlaceholderManager
import com.willfp.eco.core.placeholder.PlayerPlaceholder
import com.willfp.eco.util.StringUtils
import org.bukkit.potion.PotionEffectType

object LevelPlaceholder {
    fun register(plugin: EcoPlugin) {
        PlaceholderManager.registerPlaceholder(
            PlayerPlaceholder(
                plugin,
                "level"
            ) {

                val level: Int
                level = try {
                    PlaceholderManager.translatePlaceholders("%battlepass_tier%", it).toInt()
                } catch (e: NumberFormatException) {
                    return@PlayerPlaceholder "§80✯"
                }
                val color: String
                color = if (level == 1) {
                    "§8"
                } else if (level < 10) {
                    "§7"
                } else if (level < 20) {
                    "§f"
                } else if (level < 30) {
                    "§6"
                } else if (level < 40) {
                    "§3"
                } else if (level < 50) {
                    "§c"
                } else if (level < 60) {
                    "§e"
                } else if (level < 70) {
                    "§9"
                } else if (level < 80) {
                    "§b"
                } else if (level < 90) {
                    "§a"
                } else if (level < 100) {
                    "§d"
                } else {
                    "§a§l"
                }
                "$color$level✯"
            }
        )
        PlaceholderManager.registerPlaceholder(PlayerPlaceholder(
            plugin,
            "color"
        ) {
            PlaceholderManager.translatePlaceholders("%luckperms_prefix%", it).substring(0, 2)
        })
        PlaceholderManager.registerPlaceholder(PlayerPlaceholder(
            plugin,
            "rank"
        ) {
            val prefix = PlaceholderManager.translatePlaceholders("%luckperms_prefix%", it)
            prefix.replace(" ", "")
                .replace("|", "")
        })
        PlaceholderManager.registerPlaceholder(PlayerPlaceholder(
            plugin,
            "tag"
        ) {
            val tag = PlaceholderManager.translatePlaceholders("%deluxetags_tag%", it)
            if (tag == null || tag.isEmpty() || tag.isBlank()) {
                ""
            } else {
                " $tag"
            }
        })
        PlaceholderManager.registerPlaceholder(PlayerPlaceholder(
            plugin,
            "heart"
        ) {
            if (it.hasPotionEffect(PotionEffectType.WITHER)) {
                StringUtils.format("&0❤")
            } else if (it.hasPotionEffect(PotionEffectType.POISON)) {
                StringUtils.format("&3❤")
            } else if (it.hasPotionEffect(PotionEffectType.ABSORPTION)) {
                StringUtils.format("&6❤")
            } else {
                StringUtils.format("&c❤")
            }
        })
    }
}
